tests/utils.py

```diff
@@ -123,21 +126,50 @@ def run_server(self, repo=None, mode="local"):
         p = Popen(args)
         return p.pid
 
+    def check_pid(self):
+        """Check the 'triton start' PID and raise an exception if the process is unhealthy"""
+        # Check if the PID exists, an exception is raised if not
+        self.check_pid_with_signal()
+        # If the PID exists, check the status of the process. Raise an exception
+        # for a bad status.
+        self.check_pid_status()
+
+    def check_pid_with_signal(self):
+        """Check for the existence of a PID by sending signal 0"""
+        try:
+            proc = psutil.Process(self.pid)
+            proc.send_signal(0)
+        except psutil.NoSuchProcess as e:
+            # PID doesn't exist, passthrough the exception
+            raise e
+
+    def check_pid_status(self):
+        """Check the status of the 'triton start' process based on its PID"""
+        process = psutil.Process(self.pid)
+        # NOTE: May need to check other statuses in the future, but zombie was observed
+        # in some local test cases.
+        if process.status() == psutil.STATUS_ZOMBIE:
+            raise Exception(f"'triton start' PID {self.pid} was in a zombie state.")
+
     def wait_for_server_ready(self, timeout: int = 60):
         start = time.time()
         while time.time() - start < timeout:
             print(
-                "Waiting for server to be ready ",
+                "[DEBUG] Waiting for server to be ready ",
                 round(timeout - (time.time() - start)),
                 flush=True,
             )
             time.sleep(1)
             try:
+                print(f"[DEBUG] Checking status of 'triton start' PID {self.pid}...")
+                self.check_pid()
```

Comment:
Should sanity check that it is indeed kill_server that is hanging prior to these changes that do additional checks/communication with the 'triton start' PID.

Reply:
Investigation will be a follow-up